backend/app/main.py

```diff
@@ -54,16 +55,34 @@ def read_team(team_id: int, db: Session = Depends(get_db)):
         raise HTTPException(status_code=404, detail="Team not found")
     return db_team
 
+@app.get("/api/teams/", response_model=List[schemas.Team])
+def read_teams(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
+    teams = crud.get_teams(db, skip=skip, limit=limit)
+    return teams
+
+# Players
 @app.post("/api/players/", response_model=schemas.Player)
 def create_player(player: schemas.PlayerCreate, db: Session = Depends(get_db)):
-    return crud.create_player(db=db, player=player)
+    db_player = crud.get_player_by_firebase_id(db, firebase_id=player.firebase_id)
+    db_player_name = crud.get_player_by_name(db, name=player.name)
+    if db_player is None and db_player_name is None:
+        return crud.create_player(db=db, player=player)
+    if db_player is None:
+        raise HTTPException(status_code=404, detail="Name already used")
```

Comment:
404 means `Not found`. `400` suits better for such scenarios which is `Bad request`